src/platform/forms-system/src/js/fields/FileField.jsx

```diff
@@ -190,22 +191,25 @@ const FileField = props => {
     [formData],
   );
 
-  useEffect(() => {
-    // The File object is not preserved in the save-in-progress data
-    // We need to remove these entries; an empty `file` is included in the
-    // entry, but if API File Object still exists (within the same session), we
-    // can't use Object.keys() on it because it returns an empty array
-    const newData = files.filter(
-      // keep - file may not exist (already uploaded)
-      // keep - file may contain File object; ensure name isn't empty
-      // remove - file may be an empty object
-      data => !data.file || (data.file?.name || '') !== '',
-    );
-
-    if (newData.length !== files.length) {
-      onChange(newData);
-    }
-  });
+  useEffect(
+    () => {
+      // The File object is not preserved in the save-in-progress data
+      // We need to remove these entries; an empty `file` is included in the
+      // entry, but if API File Object still exists (within the same session), we
+      // can't use Object.keys() on it because it returns an empty array
+      const newData = files.filter(
+        // keep - file may not exist (already uploaded)
+        // keep - file may contain File object; ensure name isn't empty
+        // remove - file may be an empty object
+        data => !data.file || (data.file?.name || '') !== '',
+      );
+      if (newData.length !== files.length) {
+        onChange(newData);
+      }
+      setInitialized(true);
+    },
+    [files, onChange],
```

Comment:
Adding `setInitialized` into either `useEffect` would cause the unit test for this code to fail. Added the second argument to fix this.